File: Healthcare-app/android/MDBRealmDoctor/app/src/main/java/com/wekanmdb/storeinventory/model/encounter/Encounter.kt
package com.wekanmdb.storeinventory.model.encounter

import com.wekanmdb.storeinventory.model.appoinment.Appointment
import com.wekanmdb.storeinventory.model.embededclass.Diagnosis
import com.wekanmdb.storeinventory.model.embededclass.Encounter_Participant
import com.wekanmdb.storeinventory.model.embededclass.Reference
import com.wekanmdb.storeinventory.model.patient.Patient
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import org.bson.types.ObjectId

open class Encounter (
    @PrimaryKey var _id: ObjectId = ObjectId(),
    var identifier: String? = null,
    var status: String? = null,
    var reasonReference: Reference? = null,
    var subject: Patient? = null,
    var participant: RealmList<Encounter_Participant> = RealmList(),
    var appointment: Appointment? = null,
    var serviceProvider: Reference? = null,
    var diagnosis: Diagnosis? = null,
    var subjectIdentifier: String? = null,
    var practitionerIdentifier: String? = null,
    var nurseIdentifier: String? = null
) : RealmObject() {}

File: Healthcare-app/ios/MDBRealmPatient/MDBRealmPatient/Encounter.swift
//
//  Encounter.swift
//  MDBRealmPatient
//
//  Created by Mackbook on 07/05/22.
//

import Foundation
import RealmSwift

class Encounter: Object {
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var appointment: Appointment?
    @Persisted var diagnosis: Diagnosis?
    @Persisted var identifier: String?
    @Persisted var nurseIdentifier: String?
    @Persisted var participant: List<Encounter_Participant>
    @Persisted var practitionerIdentifier: String?
    @Persisted var reasonReference: Reference?
    @Persisted var serviceProvider: Reference?
    @Persisted var status: String?
    @Persisted var subject: Patient?
    @Persisted var subjectIdentifier: String?
}
class Diagnosis: EmbeddedObject {
    @Persisted var condition: Reference?
    @Persisted var rank: Int?
    @Persisted var use: Codable_Concept?
}
class Encounter_Participant: EmbeddedObject {
    @Persisted var individual: Reference?
    @Persisted var type: Codable_Concept?
}
